app/src/main/java/com/example/habitsmasher/MapDialog.java

```diff
@@ -0,0 +1,210 @@
+        if (mapViewBundle == null) {
+            mapViewBundle = new Bundle();
+            outState.putBundle(MAPVIEW_BUNDLE_KEY, mapViewBundle);
+        }
+
+        _mapView.onSaveInstanceState(mapViewBundle);
+    }
+
+    @Override
+    public void onAttach(@NonNull Context context) {
+        super.onAttach(context);
+        try {
+            // Get habit event fragment for later use
+            _habitEventDialog = (HabitEventDialog) getTargetFragment();
+        }
+        catch (ClassCastException e) {
+            Log.e(TAG, "Exception" + e.getMessage());
+        }
+        _fusedLocationClient = LocationServices.getFusedLocationProviderClient(context);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        _mapView.onResume();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        _mapView.onStart();
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        _mapView.onStop();
+    }
+    
+    @Override
+    public void onPause() {
+        super.onPause();
+        _mapView.onPause();
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        _mapView.onDestroy();
+    }
+
+    @Override
+    public void onLowMemory() {
+        super.onLowMemory();
+        _mapView.onLowMemory();
+    }
+
+    @Override
+    public void onMapReady(@NonNull GoogleMap googleMap) {
+        if (_selectedLocation.equals("")) {
+            getCurrentLocation(isLocationNull -> {
+                if (isLocationNull) {
+                    // location is recorded as null for some reason
+                    _habitEventDialog.displayErrorMessage(HabitEventDialog.LOCATION_NOT_RECORDED);
+                    getDialog().dismiss();
+                }
+                else {
+                   setUpMap(googleMap);
+                }
+            });
+        }
+        else {
+            setUpMap(googleMap);
+        }
+    }
+
+    /**
+     * Method that setups the Google Map snippet
+     * @param googleMap google map snippet
+     */
+    private void setUpMap(GoogleMap googleMap) {
+        String[] latLngPair = _selectedLocation.split(" ");
+        LatLng coordinate = new LatLng(Double.valueOf(latLngPair[0]),
+                Double.valueOf(latLngPair[1]));
+        googleMap.addMarker(new MarkerOptions().position(coordinate).title("Habit Event Location"));
+        googleMap.animateCamera(CameraUpdateFactory.newLatLngZoom(coordinate,
+                googleMap.getMaxZoomLevel()));
+        googleMap.setOnMapClickListener(new GoogleMap.OnMapClickListener() {
+            @Override
+            public void onMapClick(@NonNull LatLng latLng) {
+                // select location
+                googleMap.clear();
+                googleMap.addMarker(new MarkerOptions().position(latLng).title("Habit Event Location"));
+                _selectedLocation = latLng.latitude + " " + latLng.longitude;
+            }
```

Comment:
This zooms and centers the map at the specified location and places a marker at that point. The user is then able to click on a point on the map, in which it spawns a marker at that location (deleting the other).